extendeddatacrossword_test.go

```diff
@@ -420,41 +390,41 @@ func TestCorruptedEdsReturnsErrByzantineData_UnorderedShares(t *testing.T) {
 		},
 	}
 
-	for codecName, codec := range codecs {
-		t.Run(codecName, func(t *testing.T) {
-			// create a DA header
-			eds := createTestEdsWithNMT(t, codec, shareSize, namespaceSize, 1, 2, 3, 4)
-			assert.NotNil(t, eds)
-			dAHeaderRoots, err := eds.getRowRoots()
-			assert.NoError(t, err)
-
-			dAHeaderCols, err := eds.getColRoots()
-			assert.NoError(t, err)
-			for _, test := range tests {
-				t.Run(test.name, func(t *testing.T) {
-					// create an eds with the given shares
-					corruptEds := createTestEdsWithNMT(t, codec, shareSize, namespaceSize, sharesValue...)
-					assert.NotNil(t, corruptEds)
-					// corrupt it by setting the values at the given coordinates
-					for i, coords := range test.coords {
-						x := coords[0]
-						y := coords[1]
-						corruptEds.setCell(x, y, test.values[i])
-					}
+	codec := NewLeoRSCodec()
+
+	t.Run(codec.Name(), func(t *testing.T) {
+		// create a DA header
+		eds := createTestEdsWithNMT(t, codec, shareSize, namespaceSize, 1, 2, 3, 4)
+		assert.NotNil(t, eds)
+		dAHeaderRoots, err := eds.getRowRoots()
+		assert.NoError(t, err)
+
+		dAHeaderCols, err := eds.getColRoots()
+		assert.NoError(t, err)
+		for _, test := range tests {
+			t.Run(test.name, func(t *testing.T) {
+				// create an eds with the given shares
+				corruptEds := createTestEdsWithNMT(t, codec, shareSize, namespaceSize, sharesValue...)
+				assert.NotNil(t, corruptEds)
+				// corrupt it by setting the values at the given coordinates
+				for i, coords := range test.coords {
+					x := coords[0]
+					y := coords[1]
+					corruptEds.setCell(x, y, test.values[i])
+				}
 
-					err = corruptEds.Repair(dAHeaderRoots, dAHeaderCols)
-					assert.Equal(t, err != nil, test.wantErr)
-					if test.wantErr {
-						var byzErr *ErrByzantineData
-						assert.ErrorAs(t, err, &byzErr)
-						errors.As(err, &byzErr)
-						assert.Equal(t, byzErr.Axis, test.corruptedAxis)
-						assert.Equal(t, byzErr.Index, test.corruptedIndex)
-					}
-				})
-			}
-		})
-	}
+				err = corruptEds.Repair(dAHeaderRoots, dAHeaderCols)
+				assert.Equal(t, err != nil, test.wantErr)
+				if test.wantErr {
+					var byzErr *ErrByzantineData
+					assert.ErrorAs(t, err, &byzErr)
+					errors.As(err, &byzErr)
+					assert.Equal(t, byzErr.Axis, test.corruptedAxis)
+					assert.Equal(t, byzErr.Index, test.corruptedIndex)
+				}
+			})
+		}
+	})
```

Comment:
[optional] this test could remove the outer t.Run because it only tests one codec

Reply:
oh yeah nice catch, will remove it